src/core/server/saved_objects/export/get_sorted_objects_for_export.ts

```diff
@@ -185,10 +187,40 @@ export async function exportSavedObjectsToStream({
     ({ namespaces, ...object }) => object
   );
 
+  // update the credential fields from "data-source" saved object to use placeholder to avoid exporting sensitive information
+  const redactedObjectsWithoutCredentials = redactedObjects.map<SavedObject<unknown>>((object) => {
+    if (object.type === 'data-source') {
+      const { auth, ...rest } = object.attributes as {
+        auth: { type: string; credentials?: any };
+      };
+      const hasCredentials = auth && auth.credentials;
+      const updatedCredentials = hasCredentials
+        ? Object.keys(auth.credentials).reduce((acc, key) => {
+            acc[key] = DATA_SOURCE_CREDENTIALS_PLACEHOLDER;
```

Comment:
Thanks for this change~
Just one question, SigV4 and TokenExchange also have regions under credential attribute, will this logic have any side effect?

Maybe we can also test SigV4 scenario?

Reply:
every field under "credentials" attributes will be replaced to use placeholder value, which . I have tested sigV4 case, you can find it from the video. The exported sigV4 data source will look like this
```
{
   "attributes": {
      "auth": {
         "credentials": {
            "accessKey": "pleaseUpdateCredentials",
            "region": "pleaseUpdateCredentials",
            "secretKey": "pleaseUpdateCredentials",
            "service": "pleaseUpdateCredentials"
         },
         "type": "sigv4"
      },
      "dataSourceVersion": "",
      "description": "",
      "endpoint": "https://mjdty4727oifwax8ycdk.us-west-2.aoss.amazonaws.com",
      "installedPlugins": [],
      "title": "testSigV4"
   },
   "id": "7f8e13c0-2378-11ef-a345-bd55de69299f",
   "migrationVersion": {
      "data-source": "2.4.0"
   },
   "references": [],
   "type": "data-source",
   "updated_at": "2024-06-05T20:16:31.484Z",
   "version": "WzQsMV0="
}

```